controllers/nnf_node_ec_data_controller.go

```diff
@@ -0,0 +1,205 @@
+/*
```

Comment:
It seems weird to me that this code is in nnf-sos and is part of the controller framework. It's not a reconciler, and it's not part of nnf-sos. Is there something preventing you from making a k8s client down in nnf-ec? Or passing the existing k8s client from nnf-sos to nnf-ec if you wanted to re-use it?

Reply:
I was trying to preserve nnf-ec's lack of dependency on kubernetes. That way nnf-ec can continue to be an independent program.

I suppose I could pass in the k8sClient/NamespacedName and in nnf-ec implement the CRD Storage Provider... I'm not sure where the CRD would live then; and how would it get deployed?

Reply:
A CRD doesn't need to be paired with a reconciler.  There is  an option on kubebuilder/operator-sdk  to say you won't want the reconciler.

Reply:
I do think we'll eventually want code in reconciler strictly for delete handling in place of `nnf-ec --purge`. Using purge only deletes namespaces and is therefore problematic.